roles/etcd/tasks/gen_certs_vault.yml

```diff
@@ -37,6 +37,8 @@
         Accept: application/json
         Content-Type: application/json
         X-Vault-Token: "{{ hostvars[groups.etcd|first]['etcd_vault_login_result']['json']['auth']['client_token'] }}"
+  when: inventory_hostname == groups.etcd|first and
+        hostvars[groups.etcd|first].get('etcd_vault_login_result', {}).get('json', {}).get('auth', {}).get('client_token') != ""
```

Comment:
This is quite a challenging condition to read. If I read it right, it looks like you are testing if etcd[0] has a vault client token. Maybe this should be set in a separate fact accessible to all hosts (with run_once and delegate) using a `set_fact` so that it's clear what is being tested.